File: src/main/java/application/history/history/application/inputs/CreateHistoryInput.java
package application.history.history.application.inputs;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.sun.istack.NotNull;
import lombok.Getter;

import java.time.LocalDate;

@Getter
public final class CreateHistoryInput {
    @NotNull
    private final String content;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @NotNull
    private final LocalDate date;

    public CreateHistoryInput(String content, LocalDate date) {
        this.content = content;
        this.date = date;
    }
}

File: src/main/kotlin/com/example/history/application/inputs/CreateHistoryInput.kt
package com.example.history.application.inputs

import com.fasterxml.jackson.annotation.JsonFormat
import com.sun.istack.NotNull
import java.time.LocalDate

class CreateHistoryInput(@field:NotNull private val content: String, date: LocalDate) {
    fun getContent(): String {
        return this.content;
    }

    fun getDate(): LocalDate {
        return this.date;
    }

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @NotNull
    private val date: LocalDate

    init {
        this.date = date
    }
}
